### DISCUSSION TITLE
How to Enforce Sync Order in ArgoCD app of apps pattern?

### DISCUSSION THREAD

**Discussion by fjammes** [argoproj/argo-cd#19712] [Q&A]
I'm using the app of apps pattern in ArgoCD and I'm looking for a way to enforce a sync order between certain apps. Specifically, I want the strimzi app to be synced before my app containing the Kafka CR, and the minio-operator app to be synced before my app containing the Minio CR.

Is there an easy way to manage dependencies or sync order between apps in ArgoCD?

Here is a link to my argoCD configuration: https://github.com/astrolabsoftware/fink-cd/tree/v3.1.3-rc1/apps/templates

Thanks in advance!

**Comment by rumstead** [comment]
Since you are using Apps of Apps, have you tried [sync waves](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/#how-do-i-configure-waves)?

**Reply by fjammes** [reply]
I tried `SyncPhases` but it seems it works only inside an app, not for synchronizing multiple apps between them. I suppose this is the same thing for sync waves?

**Reply by rumstead** [reply]
If you are using Apps of Apps, you can use `SyncWaves` in the "first App". Eg, what is described [here](https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/1.7-1.8/#health-assessment-of-argoprojioapplication-crd-has-been-removed).

**Comment by pasha-codefresh** [comment]
You probably also want to try https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Progressive-Syncs/
It uses application set, but it should help you achieve what you need

**Comment by pasha-codefresh** [comment]
More details you can find here 
https://github.com/argoproj/argo-cd/issues/7437

**Comment by sahma19** [comment]
I would recommend using SyncWave. Lowest value syncs first.

App 1: `argocd.argoproj.io/sync-wave: "100"`
App 2: `argocd.argoproj.io/sync-wave: "200"`

**Comment by kostis-codefresh** [comment]
You need to make Argo CD aware of App dependencies as described here https://codefresh.io/blog/argo-cd-application-dependencies/
